src/main/java/org/apache/onami/persist/StatefulTransactionHook.java

```diff
@@ -0,0 +1,16 @@
+package org.apache.onami.persist;
+
+/**
+ * A transaction hook that is run at the end of a @Transactional block both preCommit() and postCommit().
+ */
+public interface StatefulTransactionHook {
```

Comment:
my only suggestion is to call this a `callback` instead of `hook`, but I don't care all that much. it's just onami after all.